runtime-sdk/src/modules/core/mod.rs

```diff
@@ -133,6 +133,19 @@ pub struct TxSimulationFailure {
     code: u32,
 }
 
+impl TxSimulationFailure {
+    /// Returns true if the failure is "core::Error::OutOfGas" or "evm::OutOfGas".
+    pub fn is_error_out_of_gas(&self) -> bool {
+        (self.module_name == MODULE_NAME && self.code == 12)
+            || (self.module_name == "evm" && self.code == 2 && self.message.contains("out of gas"))
```

Comment:
Hm this doesn't seem ok as this now relies on implementation details of the `evm` module. Can we do this differently?

We could stop using the special "evm out of gas" error and use `core::Error::OutOfGas` instead (similar to what we do in the `contracts` module). Should be simple to do in the `From<evm::ExitError>` implementation.

Reply:
Makes sense, ll do. (I was actually surprised this was not done this way already).